test/extended/node/oom_detection.go

```diff
@@ -0,0 +1,74 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	o "github.com/onsi/gomega"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: expand this test to SSH into every node in the cluster and gather kubelet logs in case kubelet doesn't send
+// 		 events to the api-server or `oc adm node-logs`
+var _ = g.Describe("[sig-node][Late]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("no-oom-kills")
+
+	g.It("should not have pods that are OOM Killed", func() {
+		kubeClient := oc.AdminKubeClient()
+
+		events, err := kubeClient.CoreV1().Events("").List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			g.Fail(fmt.Sprintf("Unexpected error: %v", err))
+		}
+		oomEventNodeList := []string{}
+		errorMsg := "System OOM encountered"
```

Comment:
You need to filter by suite start (see other event types where we filter those out).  We will eventually be running multiple suites in order - we might run disruptive (which causes OOMs) before we run e2e-conformanec (to verify that everything works after ooms).  If you look back before the suite start you may cause failures.

Reply:
Ok makes sense. Let me look at that.